bugwarrior/services/deck.py

```diff
@@ -0,0 +1,184 @@
+        },
+        CARD_ID: {
+            'type': 'numeric',
+            'label': 'Nextcloud Deck Card ID',
+        },
+        TITLE: {
+            'type': 'string',
+            'label': 'Nextcloud Deck Card Title',
+        },
+        DESCRIPTION: {
+            'type': 'string',
+            'label': 'Nextcloud Deck Card Description',
+        },
+        ORDER: {
+            'type': 'numeric',
+            'label': 'Nextcloud Deck Order',
+        },
+        ASSIGNEE: {
+            'type': 'string',
+            'label': 'Nextcloud Deck Assignee(s)',
+        }
+    }
+
+    UNIQUE_KEY = (BOARD_ID, STACK_ID, CARD_ID,)
+
+    def to_taskwarrior(self):
+        return {
+            'project': self.extra['board']['title'].lower().replace(' ', '_'),
+            'priority': self.get_priority(),
+            # 'annotations': self.record.get('annotations', []),
+            'tags': self.get_tags(),
+            'entry': datetime.datetime.fromtimestamp(self.record.get('createdAt'), tz=tzutc()),
+            'due': self.parse_date(self.record.get('duedate')),
+
+            self.AUTHOR: self.record['owner']['uid'],
+            self.BOARD_ID: self.extra['board']['id'],
+            self.BOARD_TITLE: self.extra['board']['title'],
+            self.STACK_ID: self.extra['stack']['id'],
+            self.STACK_TITLE: self.extra['stack']['title'],
+            self.CARD_ID: self.record['id'],
+            self.TITLE: self.record['title'],
+            self.DESCRIPTION: self.record['description'],
+            self.ORDER: self.record['order'],
+            self.ASSIGNEE: ([assignee['participant']['uid'] for assignee in self.record['assignedUsers']] or [None])[0],
+        }
+
+    def get_tags(self):
+        return self.get_tags_from_labels([label['title'] for label in self.record['labels']])
+
+    def get_default_description(self):
+        return self.build_default_description(title=self.record['title'])
+
+
+class NextcloudDeckService(IssueService):
+    ISSUE_CLASS = NextcloudDeckIssue
+    CONFIG_SCHEMA = NextcloudDeckConfig
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.client = NextcloudDeckClient(
+            base_uri=self.config.base_uri,
+            username=self.config.username,
+            password=self.config.password
+        )
+
+    def get_service_metadata(self):
+        return {
+            'import_labels_as_tags': self.config.import_labels_as_tags,
+            'label_template': self.config.label_template,
+            'only_if_assigned': self.config.only_if_assigned,
+        }
+
+    def get_owner(self, issue):
+        return issue[issue.ASSIGNEE]
+
+    def issues(self):
+        for board in self.client.get_boards():
+            if ((not self.config.include_board_ids) or (str(board['id']) in self.config.include_board_ids)) and \
+                    ((not self.config.exclude_board_ids) or (str(board['id']) not in self.config.exclude_board_ids)):
```

Comment:
I would like to see this complex logic replaced with a filter function. Many of the services have a method called `filter_repos` you could see for reference. (You'd probably call yours `filter_boards`.)